core/controllers/base.py

```diff
@@ -351,8 +351,10 @@ def render_template(
         if 'nav_mode' not in values:
             values['nav_mode'] = ''
 
-        if redirect_url_on_logout is None:
+        if redirect_url_on_logout is None and 'create' not in self.request.uri:
```

Comment:
So, I think this is also problematic. The "create" pages of published explorations are viewable by guests. If you log out when on the 'create' page of a public exploration you should still stay on that page, and it won't 404.

I think, only for private explorations, we'll need to change the logout link somehow to redirect to /library. But we should only do that if the exploration hasn't gotten published by the user in the same session.